crates/bevy_ecs/src/storage/resource.rs

```diff
@@ -0,0 +1,174 @@
+    #[inline]
+    pub fn get_ticks(&self) -> Option<&ComponentTicks> {
+        self.column
+            .get_ticks(0)
+            // SAFETY: If the first row exists, a valid ticks value has been written.
+            .map(|ticks| unsafe { ticks.deref() })
+    }
+
+    /// Gets a mutable reference to the change ticks of the underlying resource, if available.
+    #[inline]
+    pub fn get_ticks_mut(&mut self) -> Option<&mut ComponentTicks> {
+        self.column
+            .get_ticks(0)
+            // SAFETY: If the first row exists, a valid ticks value has been written.
+            // This function has exclusvie access to the underlying column.
+            .map(|ticks| unsafe { ticks.deref_mut() })
+    }
+
+    #[inline]
+    pub(crate) fn get_with_ticks(&self) -> Option<(Ptr<'_>, &UnsafeCell<ComponentTicks>)> {
+        self.column.get(0)
+    }
+
+    /// Inserts a value into the resource. If a value is already present
+    /// it will be replaced.
+    ///
+    /// # Safety
+    /// `value` must be valid for the underlying type for the resource.
+    #[inline]
+    pub unsafe fn insert(&mut self, value: OwningPtr<'_>, change_tick: u32) {
+        if self.is_present() {
+            self.column.replace(0, value, change_tick);
+        } else {
+            self.column.push(value, ComponentTicks::new(change_tick));
+        }
+    }
+
+    #[inline]
+    pub(crate) unsafe fn insert_with_ticks(
+        &mut self,
+        value: OwningPtr<'_>,
+        change_ticks: ComponentTicks,
+    ) {
+        if self.is_present() {
+            self.column.replace_untracked(0, value);
+            *self.column.get_ticks_unchecked(0).deref_mut() = change_ticks;
+        } else {
+            self.column.push(value, change_ticks);
+        }
+    }
+
+    /// Removes a value from the resource, if present.
+    #[inline]
+    #[must_use = "The returned pointer to the removed component should be used or dropped"]
+    pub fn remove(&mut self) -> Option<(OwningPtr<'_>, ComponentTicks)> {
+        self.column.swap_remove_and_forget(0)
+    }
+
+    #[inline]
+    pub(crate) fn remove_and_drop(&mut self) {
+        self.column.clear();
+    }
+}
+
+/// The backing store for all [`Resource`]s stored in the [`World`].
+///
+/// [`Resource`]: crate::system::Resource
+/// [`World`]: crate::world::World
+#[derive(Default)]
+pub struct Resources {
+    resources: SparseSet<ComponentId, ResourceData>,
+}
+
+impl Resources {
+    /// The total number of resoruces stored in the [`World`]
+    ///
+    /// [`World`]: crate::world::World
+    #[inline]
+    pub fn len(&self) -> usize {
+        self.resources.len()
+    }
+
+    /// Returns true if there are no resources stored in the [`World`],
+    /// false otherwise.
+    ///
+    /// [`World`]: crate::world::World
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.resources.is_empty()
+    }
+
+    /// Gets a read-only [`Ptr`] to a resource, if available.
```

Comment:
```suggestion
    /// Gets read-only access to a resource, if it exists.
```

Seems like this was refactored to not return a `Ptr` anymore, so the docs should be updated to reflect that.